File: Src/Compilers/CSharp/Source/Utilities/ArrayElement.cs
namespace Roslyn.Compilers.CSharp
{
    internal struct ArrayElement<T>
    {
        internal T Value;

        internal ArrayElement(T value)
        {
            this.Value = value;
        }

        public static implicit operator ArrayElement<T>(T value)
        {
            return new ArrayElement<T>(value);
        }
    }
}

File: Src/Compilers/VisualBasic/Source/Utilities/ArrayElement.vb
Imports Roslyn.Compilers.Internal

Namespace Roslyn.Compilers.VisualBasic
    Friend Structure ArrayElement(Of T As Class)
        Friend value As T

        Public Shared Widening Operator CType(element As ArrayElement(Of T)) As T
            Return element.value
        End Operator
    End Structure
End Namespace
